File: conveyor/src/main/java/ru/artemev/conveyor/dto/PaymentScheduleElement.java
package ru.artemev.conveyor.dto;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Builder
@Schema(description = "График платежей")
public class PaymentScheduleElement {
  @Schema(description = "Номер платежа", example = "1")
  private Integer number;

  @Schema(description = "Дата платежа", example = "2000-01-01")
  private LocalDate date;

  @Schema(description = "Всего к оплате ", example = "1000")
  private BigDecimal totalPayment;

  @Schema(description = "Выплата процентов ", example = "1000")
  private BigDecimal interestPayment;

  @Schema(description = "Выплата долга ", example = "1000")
  private BigDecimal debtPayment;

  @Schema(description = "Оставшийся долг ", example = "1000")
  private BigDecimal remainingDebt;
}

File: deal/src/main/kotlin/ru/kit/deal/dto/PaymentScheduleElement.kt
package ru.kit.deal.dto

import io.swagger.v3.oas.annotations.media.Schema
import java.math.BigDecimal
import java.time.LocalDate

@Schema(description = "График платежей")
data class PaymentScheduleElement(
    @Schema(description = "Номер платежа", example = "1")
    val number: Int? = null,

    @Schema(description = "Дата платежа", example = "2000-01-01")
    val date: LocalDate? = null,

    @Schema(description = "Всего к оплате ", example = "1000")
    val totalPayment: BigDecimal? = null,

    @Schema(description = "Выплата процентов ", example = "1000")
    val interestPayment: BigDecimal? = null,

    @Schema(description = "Выплата долга ", example = "1000")
    val debtPayment: BigDecimal? = null,

    @Schema(description = "Оставшийся долг ", example = "1000")
    val remainingDebt: BigDecimal? = null,
)